packages/sdk-react/src/core/queries/useTag.ts

```diff
@@ -1,79 +1,90 @@
 import { toast } from 'react-hot-toast';
 
-import {
-  TagCreateOrUpdateSchema,
-  TagReadSchema,
-  TagsPaginationResponse,
-  TagService,
-  ApiError,
-} from '@monite/sdk-api';
-import { useMutation, useQuery } from '@tanstack/react-query';
-
-import { useMoniteContext } from '../context/MoniteContext';
-import { useEntityListCache } from './hooks';
-
-export const TAG_QUERY_ID = 'tags';
-
-const useTagListCache = () =>
-  useEntityListCache<TagReadSchema>(() => [TAG_QUERY_ID]);
-
-export const useTagList = (...args: Parameters<TagService['getList']>) => {
-  const { monite } = useMoniteContext();
-
-  return useQuery<TagsPaginationResponse, ApiError>({
-    queryKey: [TAG_QUERY_ID, { variables: args }],
-
-    queryFn: () => monite.api.tag.getList(...args),
-  });
-};
-
-export const useCreateTag = () => {
-  const { monite } = useMoniteContext();
-  const { invalidate } = useTagListCache();
-
-  return useMutation<TagReadSchema, ApiError, TagCreateOrUpdateSchema>({
-    mutationFn: (args) => monite.api.tag.create(args),
-
-    onSuccess: () => {
-      invalidate();
-    },
-
-    onError: async (error) => {
-      toast.error(error.body.error.message || error.message);
+import { api, useMoniteApiClient } from '@/api/client';
+import { getAPIErrorMessage } from '@/utils/getAPIErrorMessage';
+import { useLingui } from '@lingui/react';
+import { useQueryClient } from '@tanstack/react-query';
+
+export const useTagListQuery = (
+  query?: (typeof api.tags.getTags.types.parameters)['query']
+) => {
+  const { api, apiVersion, entityId } = useMoniteApiClient();
+  return api.tags.getTags.useQuery({
+    header: {
+      'x-monite-version': apiVersion,
+      'x-monite-entity-id': entityId,
     },
+    query,
   });
 };
 
-export const useUpdateTag = () => {
-  const { monite } = useMoniteContext();
-  const { invalidate } = useTagListCache();
-
-  return useMutation<
-    TagReadSchema,
-    ApiError,
-    { id: string; payload: { name: string } }
-  >({
-    mutationFn: ({ id, payload }) => monite.api.tag.update(id, payload),
-
-    onSuccess: () => {
-      invalidate();
-    },
+export const useCreateTagMutation = () => {
+  const { i18n } = useLingui();
+  const { api, apiVersion, entityId } = useMoniteApiClient();
```

Comment:
unused variables `apiVersion`, `entityId`

Reply:
Thank you, I even refactored this mutation hook.